.github/workflows/client-report-workflow-related-resources.yaml

```diff
@@ -0,0 +1,194 @@
+name: Report Related Resources
+
+# This workflow is meant to be called from other repositories' workflows to report that the
+# workflow impacts or is related to some resource in Sherlock.
+#
+# The caller repository must have Workload Identity Federation configured to allow impersonation of the
+# "dsp-tools-iap-access@dsp-tools-k8s.iam.gserviceaccount.com" service account; steps 1 and 2 of the documentation:
+# https://docs.google.com/document/d/1bnhDmWQHAMat_Saoa_z28FHwXmGWw6kywjdbKf208h4/edit
+#
+# With that configured, here's an example of how you can call this workflow:
+# ```yaml
+# jobs:
+#
+#   report-related-resources:
+#     uses: broadinstitute/sherlock/.github/workflows/client-report-workflow-related-resources.yaml@main
+#     needs: <your-existing-job-id>
+#     with:
+#       chart-releases: leonardo-dev
+#     permissions:
+#       id-token: write
+# ```
+
+on:
+  workflow_call:
+    inputs:
+      app-versions:
+        required: false
+        type: string
+        description: |
+          App versions affected by the calling workflow.
+          The app versions must have already been reported to Sherlock.
+          
+          App versions should be specified by an app version selector. Some different examples:
+            - <id-integer> (`456`)
+            - <chart-selector>/<version-string> (`leonardo/v1.2.3`)
+          Multiple app versions can be reported by separating them with whitespace and/or commas.
```

Comment:
[Here's a test for such a scenario](https://github.com/broadinstitute/sherlock/blob/main/sherlock/internal/controllers/v2controllers/ci_run_test.go#L190), I honestly think scrolling through that function and reading the comments might help give you a sense of how this can all come together.

Other examples might be if you've got a monorepo that builds multiple apps with independent versions -- maybe one GitHub Action would build both apps and want to report that it affects `app-one/v1.2.3, app-two/v2.3.4`. Or maybe some very wacky GitHub Action would release the same image under different versions, say, one semver and one commit hashed, so it'd want to separately report that it affects `app-one/v1.2.3, app-one/a1b2c3d4`